### PR TITLE
Decision task attribute validation should result in decision failure, not error

### PR DESCRIPTION
**Pull request by wxing1292**

### CODE REVIEW

File: service/history/decisionTaskHandler.go [hunk 1]
```diff
@@ -307,10 +328,26 @@ func (handler *decisionTaskHandlerImpl) handleDecisionCompleteWorkflow(
...
 
 	if handler.hasUnhandledEventsBeforeDecisions {
-		handler.failDecision = true
-		handler.failDecisionCause = workflow.DecisionTaskFailedCauseUnhandledDecision.Ptr()
+		return handler.handlerFailDecision(workflow.DecisionTaskFailedCauseUnhandledDecision, "")
+	}
+
+	err := handler.attrValidator.validateCompleteWorkflowExecutionAttributes(attr)
+	if err != nil {
+		if _, ok := err.(*workflow.BadRequestError); ok {
```
Comment (shreyassrivatsan) [inline service/history/decisionTaskHandler.go]: can we make this a method, `processBadRequestError` or something, that either modifies the error if its bad request or returns it as is?

Comment (shreyassrivatsan) [reply service/history/decisionTaskHandler.go]: I guess it has to take in the fail decision reason..

File: service/history/historyEngine.go [hunk 2]
```diff
@@ -864,7 +864,10 @@ func (e *historyEngineImpl) RecordActivityTaskStarted(ctx ctx.Context,
 				return nil, &h.EventAlreadyStartedError{Message: "Activity task already started."}
 			}
 
-			msBuilder.AddActivityTaskStartedEvent(ai, scheduleID, requestID, request.PollRequest.GetIdentity())
+			_, err := msBuilder.AddActivityTaskStartedEvent(ai, scheduleID, requestID, request.PollRequest.GetIdentity())
+			if err != nil {
```
Comment (shreyassrivatsan) [inline service/history/historyEngine.go]: nit: `if err := ...`

File: service/history/historyEngine.go [hunk 3]
```diff
@@ -1937,7 +1941,10 @@ func (e *historyEngineImpl) failDecision(context workflowExecutionContext, sched
 		return nil, err
 	}
 
-	msBuilder.AddDecisionTaskFailedEvent(scheduleID, startedID, cause, nil, request.GetIdentity(), "", "", "", 0)
+	_, err = msBuilder.AddDecisionTaskFailedEvent(scheduleID, startedID, cause, nil, request.GetIdentity(), "", "", "", 0)
+	if err != nil {
```
Comment (shreyassrivatsan) [inline service/history/historyEngine.go]: nit: `if err := ...`

File: service/history/mutableStateBuilder.go [hunk 4]
```diff
@@ -3372,7 +3584,11 @@ func (e *mutableStateBuilder) ReplicateChildWorkflowExecutionTimedOutEvent(event
 	return nil
 }
 
-func (e *mutableStateBuilder) CreateActivityRetryTimer(ai *persistence.ActivityInfo, failureReason string) persistence.Task {
+func (e *mutableStateBuilder) CreateActivityRetryTimer(
+	ai *persistence.ActivityInfo,
+	failureReason string,
+) persistence.Task {
+
```
Comment (shreyassrivatsan) [inline service/history/mutableStateBuilder.go]: thanks for doing all these!

File: service/history/decisionTaskHandler.go (hunk 1 above)
Comment (wxing1292) [reply service/history/decisionTaskHandler.go]: already done

**Pull request merged**